### DISCUSSION TITLE
Help on building the BIRT runtime JAR

### DISCUSSION THREAD

**Discussion by fedemg** [eclipse-birt/birt#1489] [Q&A]
Hello everyone!

I'm currently working on an application that integrates the BIRT runtime (4.8) as a dependency, which is currently brought using Innovent Solutions package from Maven. The BIRT engine is used to generate reports within the application, which are then downloaded directly by the end users.

We want to update this package to the latest BIRT release, so my question is if there's a way of building the runtime JAR only, without building the code related to the report viewer/Eclipse plugin. Also, we'd like to build the runtime with Java 8, since our application cannot compile yet with Java 11, is this even possible?

Any pointers would be useful, thanks in advance!

**Comment by hvbtup** [comment]
You don't have to build BIRT yourself, you can use the BIRT runtime package (birt-runtime-*.zip), which you can find in the releases section.
Just add /path/to/ReportEngine/lib/* to your classpath.

BIRT 4.9 requires Java 11.
BIRT 4.14 will require Java 17.

So if you want to stick with Java 8, you also have to stick with BIRT 4.8.

**Reply by fedemg** [reply]
I'm not sure I understand what you mean by "just add /path/to/ReportEngine/lib/ to your classpath"... my application is a Maven project, so I want deploy the BIRT runtime JAR to a Nexus repository where I can use it as a dependency in my POM to build my project.

**Reply by hvbtup** [reply]
My advice was regarding a classic Java project. Unfortunately the Markdown parser messed up my text.
I meant that `/path/to/ReportEngine/lib/*` including the star must be part of the ClassPath.
Regarding Maven, this is work in (non-)progress, see #625, but apparently there is a workaround, see https://github.com/eclipse-birt/birt/issues/625#issuecomment-1730079711.

**Reply by fedemg** [reply]
Hmmm... I understand, thanks! I went through the discussion in #625 before posting, but missed this comment. Essentially it means manually deploying all the required dependencies to have a set of coordinates that can be used as a dependency inside a POM.
